deltachat-ffi/src/lib.rs

```diff
@@ -811,6 +811,23 @@ pub unsafe extern "C" fn dc_set_draft(
         .unwrap_or(())
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn dc_add_device_msg(context: *mut dc_context_t, msg: *mut dc_msg_t) -> u32 {
```

Comment:
Not a `libc::??` return type?

Reply:
well, i did just a copy+paste from dc_chat_can_send() :)

i leave it for now, maybe we can go trough the ffi and streamline this at some point for all affected functions.

Reply:
`libc::uint32_t` is deprecated in favor of `u32` in Rust. And using `u32` is better than `libc::c_uint` because JNI and NAPI don't work these C types anyway.

Keep it returning `u32` and specify it as uint32_t in C API.

Reply:
If we are going to streamline it, I would rather change all ints to int32_t, because it is equal to int32_t on all existing architectures anyway.

Reply:
thanks for explaining, @link2xt